workflow/controller/operator.go

```diff
@@ -527,15 +527,13 @@ func (woc *wfOperationCtx) setGlobalParameters(executionParameters wfv1.Argument
 		if param.Value == nil && param.ValueFrom == nil {
 			return fmt.Errorf("either value or valueFrom must be specified in order to set global parameter %s", param.Name)
 		}
-		if param.ValueFrom != nil {
-			if param.ValueFrom.ConfigMapKeyRef != nil {
-				cmValue, err := common.GetConfigMapValue(woc.controller.configMapInformer, woc.wf.ObjectMeta.Namespace, param.ValueFrom.ConfigMapKeyRef.Name, param.ValueFrom.ConfigMapKeyRef.Key)
-				if err != nil {
-					return fmt.Errorf("failed to set global parameter %s from configmap with name %s and key %s: %w",
-						param.Name, param.ValueFrom.ConfigMapKeyRef.Name, param.ValueFrom.ConfigMapKeyRef.Key, err)
-				}
-				woc.globalParams["workflow.parameters."+param.Name] = cmValue
+		if param.ValueFrom != nil && param.ValueFrom.ConfigMapKeyRef != nil {
+			cmValue, err := common.GetConfigMapValue(woc.controller.configMapInformer, woc.wf.ObjectMeta.Namespace, param.ValueFrom.ConfigMapKeyRef.Name, param.ValueFrom.ConfigMapKeyRef.Key)
+			if err != nil {
+				return fmt.Errorf("failed to set global parameter %s from configmap with name %s and key %s: %w",
+					param.Name, param.ValueFrom.ConfigMapKeyRef.Name, param.ValueFrom.ConfigMapKeyRef.Key, err)
 			}
+			woc.globalParams["workflow.parameters."+param.Name] = cmValue
 		} else {
 			woc.globalParams["workflow.parameters."+param.Name] = param.Value.String()
 		}
```

Comment:
Changed the if/else structure from

```go
if param.ValueFrom != nil {
	if param.ValueFrom.ConfigMapKeyRef != nil {
		// ...
	}
} else {
	woc.globalParams["workflow.parameters."+param.Name] = param.Value.String()
}
```

to

```go
if param.ValueFrom != nil && param.ValueFrom.ConfigMapKeyRef != nil {
	// ...
} else {
	woc.globalParams["workflow.parameters."+param.Name] = param.Value.String()
}
```

Similar to `ProcessArgs`:

https://github.com/argoproj/argo-workflows/blob/dbb1bcfbd4de3295163900509fc624fb7d363b10/workflow/common/util.go#L141-L153